designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/ParametersValidator.scala

```diff
@@ -30,4 +37,38 @@ class ParametersValidator(modelData: ModelData, scenarioPropertiesNames: Iterabl
       .flatten
   }
 
+  def validate(request: AdhocTestParametersRequest, parameters: Map[String, List[Parameter]])(
+      implicit metaData: MetaData
+  ): List[NodeValidationError] = {
+    implicit val nodeId: NodeId = NodeId("")
+    val context                 = validationContextGlobalVariablesOnly
+
+    val parameterList: List[Parameter] =
+      parameters.getOrElse(
+        request.sourceParameters.sourceId,
+        throw new IllegalStateException( // This should never happen, it would mean that API is not consistent
+          s"There is no source ${request.sourceParameters.sourceId} associated with generated test parameters"
+        )
+      )
+
+    val parameterWithExpression = parameterList.map { parameter =>
+      parameter -> request.sourceParameters.parameterExpressions.getOrElse(parameter.name, parameter.finalDefaultValue)
+    }
+
+    parameterWithExpression
+      .map { case (parameter, expression) =>
+        val validatorsCompilationResult = parameter.validators.map { validator =>
+          expressionCompiler.compileValidator(validator, parameter.name, parameter.typ, context.globalVariables)
```

Comment:
We could do some refactors around `ExpressionCompiler.compileNodeParameters` and reuse this code instead of duplicating it.

Reply:
Good call, that was easy to reuse so I've removed duplicated code and reused method `expressionCompiler.compileNodeParameters`